source/margarita/upgrading.rst

```diff
@@ -0,0 +1,467 @@
+.. _upgrading:
+
+Upgrading Margarita
+===================
+
+Before attempting to upgrade your project's Margarita version, please read `the overview document
+about Margarita upgrades
+<https://github.com/caktus/django-project-template/blob/master/docs/updates.rst>`_ first, to get a
+high level view of the changes required.
+
+This document is going to outline the steps involved in upgrading a project which is using the
+master branch of Margarita. We will be upgrading it to Margarita 1.4.0. Following this guide should
+allow you to upgrade projects in a similar scenario. Projects which are already using a pinned
+version of Margarita should be able to upgrade by following the `Margarita release notes
+<https://github.com/caktus/margarita/blob/develop/CHANGES.rst>`_
+
+This guide assumes that all services are running on one server (with a pretend IP address of
+1.2.3.4). If you have a multiple server setup, then anytime a fabric command has an ``-H`` flag,
+you'll have to repeat that command once for each IP address in your fleet.
+
+
+Prepare Repository
+------------------
+
+.. WARNING::
+   The new project template does not 'reconcile' secrets because it assumes that you have
+   committed the encrypted versions to the repo, so syncing back-and-forth is not necessary. We
+   haven't yet done that step for our project, so we need to be very careful about our secrets.
+   Make sure to sync them (by doing a deploy or running ``fab get_secrets``) before going any
+   further.
+
+
+1. Get useful files from the new project template. Clone the `Django Project Template
+   <https://github.com/caktus/django-project-template>`_ repo  into a directory adjacent to ours, so
+   we can easily copy files we need:
+
+   .. code-block:: bash
+
+      ~/dev$ git clone git@github.com:caktus/django-project-template.git
+      ~/dev$ ls
+      django-project-template myproject
+
+#. Copy some important files and directory to our project:
+
+   .. code-block:: bash
+
+      ~/dev$ mv fabfile.py fabfile.py.old
```

Comment:
Yes, I waffled about which dir I should use as the working directory. I'm going to change everything to be from myproject's directory, which will make things consistent all the way through the document and also make it easier to copy/paste commands since they won't have "myproject" in it.